lib/ansible/modules/windows/win_package.ps1

```diff
@@ -1,1329 +1,524 @@
-                bResult = DuplicateTokenEx(
-                    hToken,
-                    GENERIC_ALL_ACCESS,
-                    ref sa,
-                    (int)SECURITY_IMPERSONATION_LEVEL.SecurityIdentification,
-                    (int)TOKEN_TYPE.TokenPrimary,
-                    ref hDupedToken
-                    );
-                if(!bResult)
-                {
-                    throw new Win32Exception("Duplicate Token error #" + Marshal.GetLastWin32Error().ToString());
+                $cleanup_artifacts += $log_path
+                $install_arguments = @("/i", "`"$($local_path)`"", "/log", "`"$log_path`"", "/qn", "/norestart")
+                if ($arguments -ne $null) {
+                    $install_arguments += $arguments
                 }
-                var si = new STARTUPINFO();
-                si.cb = Marshal.SizeOf(si);
-                si.lpDesktop = "";
-                bResult = CreateProcessAsUser(
-                    hDupedToken,
-                    null,
-                    strCommand,
-                    ref sa, 
-                    ref sa,
-                    false, 
-                    0, 
-                    IntPtr.Zero,
-                    null, 
-                    ref si, 
-                    ref pi
-                    );
-                if(!bResult)
-                {
-                    throw new Win32Exception("Create process as user error #" + Marshal.GetLastWin32Error().ToString());
+                $install_arguments = $install_arguments -join " "
+            } else {
+                $log_path = $null                
+                $install_exe = $local_path
+                if ($arguments -ne $null) {
+                    $install_arguments = $arguments
+                } else {
+                    $install_arguments = ""
                 }
+            }
 
-                int status = WaitForSingleObject(pi.hProcess, -1);
-                if(status == -1)
-                {
-                    throw new Win32Exception("Wait during create process failed user error #" + Marshal.GetLastWin32Error().ToString());
+            if (-not $check_mode) {
+                $process_result = Run-Process -executable $install_exe -arguments $install_arguments
+                
+                if (($log_path -ne $null) -and (Test-Path -Path $log_path)) {
+                    $log_content = Get-Content -Path $log_path -Raw
+                } else {
+                    $log_content = $null
                 }
 
-                bResult = GetExitCodeProcess(pi.hProcess, out ExitCode);
-                if(!bResult)
-                {
-                    throw new Win32Exception("Retrieving status error #" + Marshal.GetLastWin32Error().ToString());
-                }
-            }
-            finally
-            {
-                if (pi.hThread != IntPtr.Zero)
-                {
-                    CloseHandle(pi.hThread);
+                $result.exit_code = $process_result.rc
```

Comment:
I think this will be the same as `reboot_required` as `exit_code` is currently being required. I did come across that quirk which is why I just went with the original result. In the end I'll do the same as `reboot_required` where both will be kept and will deprecate `exit_code` in a future version